src/compiler/emitter.ts

```diff
@@ -2287,21 +2299,21 @@ namespace ts {
         //
 
         function emitCaseClause(node: CaseClause) {
-            writeKeyword("case");
+            emitTokenWithComment(SyntaxKind.CaseKeyword, node.pos, writeKeyword, node);
             writeSpace();
             emitExpression(node.expression);
-            writePunctuation(":");
 
-            emitCaseOrDefaultClauseStatements(node, node.statements);
+            emitCaseOrDefaultClauseStatements(node, node.statements, node.expression.end);
         }
 
         function emitDefaultClause(node: DefaultClause) {
-            writeKeyword("default");
-            writePunctuation(":");
-            emitCaseOrDefaultClauseStatements(node, node.statements);
+            const pos = emitTokenWithComment(SyntaxKind.DefaultKeyword, node.pos, writeKeyword, node);
+            emitCaseOrDefaultClauseStatements(node, node.statements, pos);
         }
 
-        function emitCaseOrDefaultClauseStatements(parentNode: Node, statements: NodeArray<Statement>) {
+        function emitCaseOrDefaultClauseStatements(parentNode: Node, statements: NodeArray<Statement>, colonPos: number) {
+            // This also handles the colon token from the preceeding clause, as weather comments must be emitted for it depends on

```

Comment:
nit: This comment is hard to read and has misspellings ('whether' not 'weather'), can you rephrase it?

Reply:
You can probably just remove the comment entirely if you rename the function to `emitCaseOrDefaultClauseRest` (as it no longer would imply only emitting the statements).